mrtt-ui/src/components/ProjectDetailsForm.js

```diff
@@ -61,26 +63,12 @@ const ProjectDetailsForm = () => {
    The api casts them to boolean so we support both */
   const showEndDateInput = watchHasProjectEndDate === 'true' || watchHasProjectEndDate === true
 
-  useEffect(
-    function initializeFormWithApiData() {
-      if (resetForm && registrationAnswersUrl) {
-        axios
-          .get(registrationAnswersUrl)
-          .then(({ data }) => {
-            const initialValuesForForm = formatApiAnswersForForm({
-              apiAnswers: data,
-              questionMapping: questionMapping.projectDetails
-            })
-
-            resetForm(initialValuesForForm)
-          })
-          .catch(() => {
-            toast.error(language.error.apiLoad)
-          })
-      }
-    },
-    [registrationAnswersUrl, resetForm]
-  )
+  usePopulateQuestionFormWithInitialValues({
+    apiUrl: registrationAnswersUrl,
+    resetForm,
+    questionMapping: questionMapping.projectDetails,
+    setIsLoading
+  })
```

Comment:
I think abstracting this to a custom react hook may be a bad code smell. Would it make sense to create a async utility function called `initializeProjectDetailsForm` that had arguments url, questionMapping, callback (cb) and onError. Then you can keep the useEffect here?

```js
useEffect(() => {
    const callback = args => {
        resetForm(args)
        setLoading(false)
    }
    onError = args => {
        toast('args words words words')
    }

    initializeProjectDetailsForm(registrationAnswersUrl, questionMapping.projectDetails, callback, onError)


},[])
```

Argument here stems from that a custom react hook that doesn't return data feels like a react component that doesn't ever return a DOM element.

Reply:
also confused by this, are we just using this to fetch initial values from the api?

Reply:
fetch, format data into form
error toast
loading indicators